### DISCUSSION TITLE
Problems with preselected dropdown

### DISCUSSION THREAD

**Discussion by DonCamillo11** [livewire/livewire#7944] [1. Help] [answered]
Hi,
I have two dropdowns, one for the client and one for the address.
Now I want to set one specific address as selected in the dropdown when changing the client dropdown.

These are the dorpdowns:

```
<select wire:model.live="client_id" name="client_id" >
  <option value="0">{{ __('Select') }}</option>
    @foreach ($clients as $client)
      <option value="{{ $client->id }}" wire:key="{{ $client->id }}">{{ $client->customername }}</option>
    @endforeach
</select>
```

```
<select wire:model="address_id">
  <option value="0">{{ __('Select') }}</option>
  @foreach ($addresses as $address)
    <option value="{{ $address->id }}" wire:key="{{ $address->id }}">{{ $address->fulladdress }}</option>
   @endforeach
</select>
```

and this is my component code
```
public function updatedClientId() {
        $this->addresses = Address::where('client_id', $this->client_id)->get();
        $this->address_id = $this->addresses->pluck('id')->first();
    }
```

All is working fine on the first change of the client dropdown, but when changing a second time an on its only updating the addresses but not the preselected option. What’s wrong here?

**Comment by joshhanley** [comment, marked as answer]
@DonCamillo11 try putting a `wire:key` on the dependant select, like `<select wire:model="address_id" wire:key="addresses-for-{{ $client_id }}>`

Hope this helps!

**Reply by DonCamillo11** [reply]
@joshhanley Awesome, thank you. I made a mistake and added the key to the `<option>`

**Reply by joshhanley** [reply]
@DonCamillo11 keys on the option is also required for the options inside the loop. What you had on the option was perfect, so I would still leave that there too 🙂
